crt_portal/cts_forms/views.py

```diff
@@ -1617,8 +1618,32 @@ def post(self, request):
 
 
 class SavedSearchActionView(LoginRequiredMixin, View):
+
     form = SavedSearchActions
 
+    def is_group_admin(self, user, group):
+
+        if not hasattr(group.group_preferences, 'admins'):
+            return False
+        admins = group.group_preferences.admins.filter(username=user.username)
+        if user in admins:
+            return True
+        return False
+
+    def get_group_data(self, user, id):
+        group_data = []
+        group_notification_preference = 'none'
+        for group in Group.objects.all():
+            if hasattr(group, 'group_preferences') and self.is_group_admin(user, group):
+                group_notification_preference = group.group_preferences.saved_searches.get(str(id), 'none')
+                group_data.append({
+                    'group': group,
+                    'notification_preferences': group_notification_preference,
+                    'field_name': f'group_{group.id}_saved_search_{id}',
+                    'notification_choices': NOTIFICATION_PREFERENCE_CHOICES['group_saved_search'],
+                })
```

Comment:
nit: reduce nesting with a guard clause here

```suggestion
            if not hasattr(group, 'group_preferences') or not self.is_group_admin(user, group):
                continue
            group_notification_preference = group.group_preferences.saved_searches.get(str(id), 'none')
            group_data.append({
                'group': group,
                'notification_preferences': group_notification_preference,
                'field_name': f'group_{group.id}_saved_search_{id}',
                'notification_choices': NOTIFICATION_PREFERENCE_CHOICES['group_saved_search'],
            })
```